example/thirdparty/ollama-js/build.mill

```diff
@@ -0,0 +1,318 @@
+
+    os.call("node_modules/.bin/unbuild", cwd = T.dest)
+
+    // prepare bundled ouptut to be used as `unmanagedDeps` in examples section.
+    os.makeDir.all(T.dest / "ollama")
+    os.copy(T.dest / "package.json", T.dest / "ollama" / "package.json")
+    os.copy(T.dest / "dist", T.dest / "ollama" / "dist")
+
+    PathRef(T.dest / "ollama")
+  }
+}
+
+trait ExampleModule extends TypeScriptModule {
+  def sources = Task.Sources(moduleDir)
+  def mainFilePath = Task { compile().path / mainFileName() }
+}
+
+trait EsmModule extends ExampleModule {
+  def enableEsm = true
+}
+
+trait Es6Module extends ExampleModule {
+  def compilerOptions =
+    super.compilerOptions() + (
+      "module" -> ujson.Str("nodenext"),
+      "moduleResolution" -> ujson.Str("nodenext"),
+      "target" -> ujson.Str("es6")
+    )
+}
+
+object `package` extends RootModule with OllamaModule with TsLintModule {
+  def moduleName = "core"
+  def npmDeps = Seq("whatwg-fetch@3.6.20")
+  def npmDevDeps = Seq(
+    "@swc/core@1.10.12",
+    "@types/whatwg-fetch@0.0.33",
+    "unbuild@2.0.0"
+  )
+
+  def npmLintDeps = Seq(
+    "prettier@3.2.4",
+    "eslint@8.29.0",
+    "@typescript-eslint/eslint-plugin@5.42.1",
+    "@typescript-eslint/parser@5.42.1"
+  )
+
+  // disable declarations output in compile stage
+  def runTypeCheck: T[Boolean] = false
+
+  object test extends TypeScriptTests with TestModule.Vitest
+
+  object examples extends TypeScriptModule {
+    object abort extends ExampleModule {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"abort-all-requests.ts"
+    }
+
+    object `fill-in-middle` extends EsmModule {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"fill.ts"
+    }
+
+    object multimodal extends EsmModule {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"multimodal.ts"
+    }
+
+    object `pull-progress` extends EsmModule {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"pull.ts"
+    }
+
+    object structured_outputs extends Es6Module {
+      def unmanagedDeps = Seq(unbuild())
+      def npmDeps = Seq("zod@3.24.1", "zod-to-json-schema@3.24.1")
+      def mainFileName = s"structured-outputs.ts"
+    }
+
+    object tools extends Es6Module {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"calculator.ts"
+    }
+  }
+}
+
+//// The Ollama JavaScript library provides the easiest way to integrate
+//// your JavaScript project with Ollama. https://github.com/ollama/ollama.
+//
+//// Project home: https://github.com/ollama/ollama-js
+
+//// Project Layout:
+
+//// SNIPPET:BUILD
+// [source,text]
+// ----
+// <project-root>
+// examples/
+// src/
+// test/
+// package.json
+// tsconfig.json
+//// SNIPPET:END
+
+// Mill will auto-generate configuration files like `tsconfig.json` and `package.json`,
+// but if the files are provided, mill would use the user provided json files instead.
```

Comment:
This part is weird. We should either have Mill be the single source of truth and generate the JSON files, or have the JSON files be the single source of truth that Mill reads. I don't have a strong opinion on which one to choose, but we should choose one or the other

Reply:
hmm ok, from mill then.